src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,145 +1,332 @@
+        if (size == elements.length) {
+            resizeArray();
+        }
+        elements[size++] = t;
+        return true;
     }
 
     @Override
-    public boolean remove(Object o) {
-        //TODO
+    public boolean remove(Object t) {
+        if(!contains(t)){
+          return false;
+        }
+        int index = this.indexOf(t);
+        if (index >= 0) {
+            remove(index);
+            return true;
+        }
         return false;
     }
 
     @Override
     public void clear() {
-        //TODO
+        for (int i = 0; i < size; i++) {
+            elements[i] = null;
+        }
+        size = 0;
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public Student get(int i) {
-        //TODO
-        return null;
+    public T get(int i) {
+        if (i >= 0 && i < size) {
+            return (T) elements[i];
+        } else {
+            throw new IndexOutOfBoundsException("Insertion index was out of range. Must be non-negative and less than or equal to size");
+        }
     }
 
+    @SuppressWarnings("unchecked")
     @Override
-    public Student set(int i, Student student) {
-        //TODO
-        return null;
+    public T set(int i, T t) {
+        if (i >= 0 && i < size) {
+            return (T) (elements[i] = t);
+        } else {
+            throw new IndexOutOfBoundsException("Insertion index was out of range. Must be non-negative and less than or equal to size");
+        }
     }
 
     @Override
-    public void add(int i, Student student) {
-        //TODO
+    public void add(int i, T t) {
+        if (i < 0 || i > size) {
+            throw new IndexOutOfBoundsException("Insertion index was out of range. Must be non-negative and less than or equal to size");
+        }
+        if (size == elements.length) {
+            resizeArray();
+        }
+        if (size - i >= 0) {
+            System.arraycopy(elements, i, elements, i + 1, size - i);
+        }
+        elements[i] = t;
+        size++;
     }
 
     @Override
-    public Student remove(int i) {
-        //TODO
-        return null;
+    public T remove(int i) {
+        if (i < 0 || i >= size) {
+            throw new IndexOutOfBoundsException("Insertion index was out of range. Must be non-negative and less than or equal to size");
+        }
+        T elementToRemove = this.get(i);
+        System.arraycopy(elements, i + 1, elements, i, size() - i);
+        elements[size] = null;
+        size--;
+        return elementToRemove;
     }
 
     @Override
-    public int indexOf(Object o) {
-        //TODO
-        return 0;
+    public int indexOf(Object t) {
+        for (int i = 0; i < size; i++) {
+            if (Objects.equals(t, elements[i])) {
+                return i;
+            }
+        }
+        return -1;
     }
 
     @Override
-    public int lastIndexOf(Object o) {
-        //TODO
-        return 0;
+    public int lastIndexOf(Object t) {
+        for (int i = 0; i < size; i++) {
```

Comment:
what is the difference between this method and indexOf ?

Reply:
indexOf - give us the first element that it found our list
lastIndexOf - give us the last element that it found our list

Reply:
Modified.